frontend/ui/widget/focusmanager.lua

```diff
@@ -115,6 +118,10 @@ function FocusManager:_verticalStep(dy)
             x = self.selected.x
             while not self.layout[self.selected.y + dy][x] do
                 x = x + 1
+                if x>100 then
+                    logger.dbg("[FocusManager] : Something went very wrong")
```

Comment:
Also we might want to make it say "too many steps without result" or something a little bit more descriptive like that.

Reply:
Hum actually, it really should never happen in normal operation. Only when changing the layout, we can end up with a malformed layout.

The best would be to crash koreader here, because in that state, it will crash not long after.

Or should I add check before to be sure it doesn't happen and continue execution ?

Reply:
You can crash, but then please do it with dbg:guard or dbg.is_on.

Reply:
I forgot, there's also the simpler `dbg.dassert()`.

https://github.com/koreader/koreader/blob/7d2ed4c3d0f61330a92ecf5a944eeedbd91c45be/frontend/dbg.lua#L50-L53

So you could write something like
```lua
if dbg.dassert(x>100) then -- will crash in debug
  logger.warn("onoez!"
end
```

No need to stick to that suggestion. But there are at least three ways to tackle the problem using `dbg`. :-)

Reply:
So I fix the problem by checking it first. So now it should print an error message and not move the cursor.